src/OSPSuite.Core/Domain/Services/SimulationConfigurationValidator.cs

```diff
@@ -16,20 +16,23 @@ class SimulationConfigurationValidator : ISimulationConfigurationValidator
       public ValidationResult Validate(SimulationConfiguration simulationConfiguration)
       {
          var validationResult = new ValidationResult();
-         //TODO
-         var module = simulationConfiguration.Module;
-         validateBuildingBlockWithFormulaCache(module?.Molecule, validationResult);
-         validateBuildingBlockWithFormulaCache(module?.Reaction, validationResult);
-         validateBuildingBlockWithFormulaCache(module?.SpatialStructure, validationResult);
-         validateBuildingBlockWithFormulaCache(module?.PassiveTransport, validationResult);
-         validateBuildingBlockWithFormulaCache(module?.Observer, validationResult);
-         validateEventGroupBuildingBlock(module?.EventGroup, module?.Molecule, validationResult);
-         module?.MoleculeStartValuesCollection.Each(cm => validateBuildingBlockWithFormulaCache(cm, validationResult));
-         module?.ParameterStartValuesCollection.Each(cm => validateBuildingBlockWithFormulaCache(cm, validationResult));
+         simulationConfiguration.ModuleConfigurations.Each(x => validateModule(x.Module, validationResult));
          simulationConfiguration.AllCalculationMethods.Each(cm => validateBuildingBlockWithFormulaCache(cm, validationResult));
          return validationResult;
       }
 
+      private void validateModule(Module module, ValidationResult validationResult)
+      {
+         validateBuildingBlockWithFormulaCache(module.Molecule, validationResult);
+         validateBuildingBlockWithFormulaCache(module.Reaction, validationResult);
+         validateBuildingBlockWithFormulaCache(module.SpatialStructure, validationResult);
+         validateBuildingBlockWithFormulaCache(module.PassiveTransport, validationResult);
+         validateBuildingBlockWithFormulaCache(module.Observer, validationResult);
+         validateEventGroupBuildingBlock(module.EventGroup, module?.Molecule, validationResult);
```

Comment:
Why "?." in `module?.Molecule` ?

Reply:
Yeah, that's probably just an oversight. The null-propagation was removed in all the first parameters, but this one had a second parameter with null-prop

Reply:
yeah I did not see it when I replaced